module4/homework/appendNewRecipe/AppendNewRecipe.cpp

```diff
@@ -0,0 +1,57 @@
+#include "AppendNewRecipe.hpp"
+
+#include <algorithm>
+#include <fstream>
+#include <map>
+
+bool AppendNewRecipe(const std::vector<std::string>& steps,
+                     const std::list<std::string>& ingredients,
+                     const std::deque<std::pair<size_t, char>>& amount) {
+    std::ofstream file("recipes.txt");
+    if (file.is_open()) {
+        std::stringstream ss = FormatRecipit(steps, ingredients, amount);
+        file << ss.str();
+
+        file.close();
+        return true;
+    }
+    return false;
+}
+
+std::vector<std::string> FormatIngredients(const std::list<std::string>& ingredients,
+                                           const std::deque<std::pair<size_t, char>>& amount) {
+    std::map<char, std::string> dimensions{{'g', "gram"}, {'s', "szklanka(i)"}, {'m', "mililitrow"}};
+
+    std::vector<std::string> vec;
+    std::transform(ingredients.begin(), ingredients.end(), amount.begin(), std::back_inserter(vec),
+                   [&dimensions](const auto& ingr, const auto& amo) {
+                       std::string str = std::to_string(amo.first);
```

Comment:
Maybe lets use std::stringstream. As far as I remember, Mateusz said that this code will run faster, because adding "strings" requires many operations